debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -195,6 +194,32 @@ public synchronized void start(Map<String, String> props) {
         }
     }
 
+    /**
+     * Builds a configuration from the given properties and validates it.
+     */
+    private Configuration buildConfiguration(Map<String, String> props) {
+        final Configuration config = Configuration.from(props);
+        boolean isValid = true;
+
+        if (!config.validateAndRecord(MySqlConnectorConfig.ALL_FIELDS, logger::error)) {
+            isValid = false;
+        }
+
+        String serverName = config.getString(MySqlConnectorConfig.SERVER_NAME);
+        String historyTopicName = config.getString(KafkaDatabaseHistory.TOPIC);
+
+        if (Objects.equals(serverName, historyTopicName)) {
+            isValid = false;
+            logger.error("The '" + MySqlConnectorConfig.SERVER_NAME.name() + "'" + " and " + "'" + KafkaDatabaseHistory.TOPIC.name() + "' values are invalid: Must not have the same value" );
+        }
+
+        if (!isValid) {
+            throw new ConnectException("Error configuring an instance of " + getClass().getSimpleName() + "; check the logs for details");
+        }
+
+        return config;
+    }
+
```

Comment:
This needs to be done within the validation phase, via methods like [those at the bottom of MySqlConnectorConfig](https://github.com/debezium/debezium/blob/master/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java#L808) that are actually used within the [field definition](https://github.com/debezium/debezium/blob/master/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java#L443).